deps/build.jl

```diff
@@ -199,6 +199,36 @@ else # source build
    cd(wdir)
 end
 
+# add shell scripts that startup another julia for some lib4ti2 programs
+# singular and libsingular are supposed to at least look in
+#  $prefixpath/lib/singular/MOD
+
+mkpath("$prefixpath/lib/singular/MOD")
+
+open("$prefixpath/lib/singular/MOD/graver.sh", "w") do io
+    write(io, "julia -e 'import lib4ti2_jll\n")
+    write(io, "lib4ti2_jll.zsolve() do x\n")
+    write(io, "p=run(ignorestatus(`\$x -G $ARGS`))\n")
+    write(io, "exit(p.exitcode)\n")
+    write(io, "end' -- \"\$@\"")
+end
```

Comment:
- IMHO the code is more readable and easier to maintain with a triple quoted string
- shell scripts should always start with a shebang (`#!...`) for maximal compatibility
- avoid Julia startup files (they just loose us time and may introduce issues)
- reduce optimizations (we just want to launch another process, no point in Julia wasting time on optimizations)
- are you sure Singular finds it if the scripts is called `graver.sh` instead of `graver`? Because the Singular interface to 4ti2 just does this:
```
   string s_name=system("executable","graver");
   if (size(s_name)==0) { s_name=system("executable","4ti2-graver"); /* debian*/ }
```

So, all in all, I get the suggested changes below (possible the `.sh` should be dropped, and of course the other scripts below also need to be adjusted)
```suggestion
write("$prefixpath/lib/singular/MOD/graver.sh", """
#!/bin/sh
julia --startup-file=no -O0 -e 'import lib4ti2_jll
lib4ti2_jll.zsolve() do x
  p=run(ignorestatus(`\$x -G $ARGS`))
  exit(p.exitcode)
end' -- "\$@"
""")
```

That leaves one major issue: in general, `lib4ti2_jll` won't be installed in the main environment: If the user just did  `add Singular` or `add Oscar`, then I think doing `import lib4ti2_jll` like that won't work (or perhaps I am missing something?).

But I think we could solve this as follows: we setup a private Julia environment for use by these scripts; that environment just has `lib4ti2_jll` set up; it could e.g. live in `Singular.jl/deps/4ti2_env` (we could either commit the relevant `Project.toml` and `Manifest.toml`, or generate them in `build.jl`). We'd then insert the path to the environment intro the `graver` etc. scripts as we generate them.